tests/run-make/amdgpu-kd/rmake.rs

```diff
@@ -0,0 +1,17 @@
+// Checks that the produced object has the .kd symbol exported.
```

Comment:
Nit: can you include slightly more context here to help a future person if this test ever fails, e.g. just ur PR description or sth

> The host runtime (HIP or HSA) expects a kernel descriptor object for each kernel in the ELF file. The amdgpu LLVM backend generates the object. It is created as a symbol with the name of the kernel plus a `.kd` suffix.